src/app/icd/ICDManager.cpp

```diff
@@ -204,5 +219,16 @@ void ICDManager::OnActiveModeDone(System::Layer * aLayer, void * appState)
         pIcdManager->UpdateOperationState(OperationalState::IdleMode);
     }
 }
+
+void ICDManager::OnTransitionToIdle(System::Layer * aLayer, void * appState)
+{
+    ICDManager * pIcdManager = reinterpret_cast<ICDManager *>(appState);
+
+    // OnTransitionToIdle will trigger a report message if reporting is needed, which should extend the active mode until the
+    // ack for the report is received.
+    pIcdManager->mStateObserver->OnTransitionToIdle();
```

Comment:
since this is async, I think this is correct. but setting mTransitionToIdleCalled just before would have been safer